CRM/Core/BAO/FinancialTrxn.php

```diff
@@ -380,18 +376,37 @@ public static function createPremiumTrxn($params) {
    * @param array $params
    *   To create trxn entries.
    *
-   * @return bool|void
+   * @throws \CRM_Core_Exception
    */
   public static function recordFees($params) {
-    $domainId = CRM_Core_Config::domainID();
-    $amount = 0;
+    // @todo: These two checks are extracted from CRM_Contribute_BAO_Contribution::recordFinancialAccounts() and should be simplified
+    //   This is the initial "no change" extraction.
+    if (($params['contribution_status_id'] != CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'contribution_status_id', 'Failed'))
+      && (!($params['contribution_status_id'] == CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'contribution_status_id', 'Pending')
+        && !$params['contribution']->is_pay_later))
+    ) {
+      // Record the fees
+    }
+    else {
+      return;
+    }
+    if (!empty($params['fee_amount'])
+      && (empty($params['prevContribution']) || $params['contribution']->fee_amount != $params['prevContribution']->fee_amount)
+    ) {
+      // Record the fees
+    }
+    else {
+      return;
+    }
+
```

Comment:
These "odd" if statements are a result of copying from the parent function and will be refactored later.